### PR TITLE
refactor: replace old default env with getRelayEnv()

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [] 

### Description

Replacing instances of the old relay env `defaultEnvironment -> getRelayEnvironment()`

The reason for this is that the prev one was creating weird UI updating issues like [this](https://github.com/artsy/eigen/pull/8391) and [this](https://github.com/artsy/eigen/pull/8219)

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-  replace old default env with getRelayEnv - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Scenes/Artwork/Artwork.tsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ import { ArtistSeriesMoreSeriesFragmentContainer as ArtistSeriesMoreSeries } fro
 import { OfferSubmittedModal } from "app/Scenes/Inbox/Components/Conversations/OfferSubmittedModal"
 import { GlobalStore } from "app/store/GlobalStore"
 import { navigationEvents } from "app/system/navigation/navigate"
+
 import { defaultEnvironment } from "app/system/relay/createEnvironment"
```
Comment (MrSltun) [inline src/app/Scenes/Artwork/Artwork.tsx:16]: 👀

File: src/app/Scenes/Artwork/Artwork.tsx [hunk 2]
```diff
@@ -12,6 +12,7 @@ import { ArtistSeriesMoreSeriesFragmentContainer as ArtistSeriesMoreSeries } fro
 import { OfferSubmittedModal } from "app/Scenes/Inbox/Components/Conversations/OfferSubmittedModal"
 import { GlobalStore } from "app/store/GlobalStore"
 import { navigationEvents } from "app/system/navigation/navigate"
+
```
Comment (dimatretyak) [inline src/app/Scenes/Artwork/Artwork.tsx:15]: do we need it here?

**Review by MrSltun — APPROVED** [review]
Looks great! 🙌 ✨

**Review by dimatretyak — DISMISSED** [review]
LGTM 👍

File: src/app/Scenes/Artwork/Artwork.tsx (hunk 1 above)
Comment (gkartalis) [reply src/app/Scenes/Artwork/Artwork.tsx:16]: 🫢

File: src/app/Scenes/Artwork/Artwork.tsx (hunk 2 above)
Comment (gkartalis) [inline src/app/Scenes/Artwork/Artwork.tsx:15]:
```suggestion
```

File: src/app/Scenes/Artwork/Artwork.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/app/Scenes/Artwork/Artwork.tsx]: it will be replace after rebasing [this changes](https://github.com/artsy/eigen/pull/8391)

**Review by dimatretyak — APPROVED** [review]

File: src/app/Scenes/Artwork/Artwork.tsx [hunk 3]
```diff
@@ -608,7 +609,7 @@ export const ArtworkQueryRenderer: React.FC<{
         render={() => {
           return (
             <AboveTheFoldQueryRenderer<ArtworkAboveTheFoldQuery, ArtworkBelowTheFoldQuery>
-              environment={environment || getRelayEnvironment()}
+              environment={environment || defaultEnvironment}
```
Comment (dimatretyak) [inline src/app/Scenes/Artwork/Artwork.tsx:612]: do we need to use `getRelayEnvironment`?

Comment (gkartalis) [reply src/app/Scenes/Artwork/Artwork.tsx:612]: oh just rebased 🤔

**Review by dimatretyak — APPROVED** [review]

**Pull request merged**